builder/builder_windows.go

```diff
@@ -0,0 +1,320 @@
+	err := task.Dag.RemoveEdge(parent.Name, child.Name)
+	if err != nil {
+		errorChan <- errors.Wrap(err, "failed to remove edge")
+		return
+	}
+
+	degree := child.GetDegree()
+	if degree == 0 {
+		step := child.Value
+		err := b.runStep(ctx, step)
+		if err != nil && step.IgnoreErrors {
+			log.Printf("Step id: %s encountered an error: %v, but is set to ignore errors. Continuing...\n", step.ID, err)
+			step.StepStatus = graph.Successful
+			for _, c := range child.Children() {
+				go b.processVertex(ctx, task, child, c, errorChan)
+			}
+		} else if err != nil {
+			step.StepStatus = graph.Failed
+			errorChan <- errors.Wrapf(err, "failed to run step id: %s", step.ID)
+		} else {
+			step.StepStatus = graph.Successful
+			for _, c := range child.Children() {
+				go b.processVertex(ctx, task, child, c, errorChan)
+			}
+		}
+		// Step must always be marked as complete.
+		step.CompletedChan <- true
+	}
+}
+
+func (b *Builder) runStep(ctx context.Context, step *graph.Step) error {
+	log.Printf("Executing step: %s\n", step.ID)
+	if step.StartDelay > 0 {
+		log.Printf("Waiting %d seconds before executing step ID: %s\n", step.StartDelay, step.ID)
+		time.Sleep(time.Duration(step.StartDelay) * time.Second)
+	}
+
+	step.StepStatus = graph.InProgress
+	step.StartTime = time.Now()
+	defer func() {
+		step.EndTime = time.Now()
+	}()
+
+	var args []string
+
+	if step.IsBuildStep() {
+		dockerfile, dockerContext := parseDockerBuildCmd(step.Build)
+		volName := b.workspaceDir
+
+		log.Println("Obtaining source code and scanning for dependencies...")
+		timeout := time.Duration(scrapeTimeoutInSec) * time.Second
+		scrapeCtx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		deps, err := b.scrapeDependencies(scrapeCtx, volName, step.WorkingDirectory, step.ID, dockerfile, dockerContext, step.Tags, step.BuildArgs)
+		if err != nil {
+			return errors.Wrap(err, "failed to scan dependencies")
+		}
+		log.Println("Successfully obtained source code and scanned for dependencies")
+		step.ImageDependencies = deps
+
+		workingDirectory := step.WorkingDirectory
+		// Modify the Run command if it's a tar or a git URL.
+		if !util.IsLocalContext(dockerContext) {
+			// NB: use step.ID as the working directory if the context is remote,
+			// since we obtained the source code from the scanner and put it in this location.
+			// If the remote context also has additional context specified, we have to append it
+			// to the working directory.
+			if util.IsGitURL(dockerContext) || util.IsVstsGitURL(dockerContext) {
+				workingDirectory = step.ID + "/" + getContextFromGitURL(dockerContext)
+			} else {
+				workingDirectory = step.ID
+			}
+
+			step.Build = replacePositionalContext(step.Build, ".")
+		}
+		// Needs hyperv isolation for older version windows images
+		args = b.getDockerRunArgs(volName, workingDirectory, step, nil, "", "docker build --isolation hyperv "+step.Build)
```

Comment:
`isolation` is already provided from `build.go` - don't default it here. If people build with Windows in a task, we should default it during step creation as a more sane default, otherwise just leave it off.